src/Doctrine/Orm/Filter/IriSearchFilter.php

```diff
@@ -0,0 +1,91 @@
+
+namespace ApiPlatform\Doctrine\Orm\Filter;
+
+use ApiPlatform\Doctrine\Common\Filter\IriSearchFilterTrait;
+use ApiPlatform\Doctrine\Orm\Util\QueryNameGeneratorInterface;
+use ApiPlatform\Metadata\OpenApiParameterFilterInterface;
+use ApiPlatform\Metadata\Operation;
+use ApiPlatform\Metadata\ParameterProviderFilterInterface;
+use ApiPlatform\Metadata\PropertiesFilterInterface;
+use ApiPlatform\State\Provider\IriConverterParameterProvider;
+use Doctrine\DBAL\Types\Types;
+use Doctrine\ORM\QueryBuilder;
+use Doctrine\Persistence\ManagerRegistry;
+use Psr\Log\LoggerInterface;
+use Symfony\Component\Serializer\NameConverter\NameConverterInterface;
+
+class IriSearchFilter extends AbstractFilter implements OpenApiParameterFilterInterface, PropertiesFilterInterface, ParameterProviderFilterInterface
+{
+    use IriSearchFilterTrait;
+
+    public function __construct(
+        ?ManagerRegistry $managerRegistry = null,
+        ?LoggerInterface $logger = null,
+        ?array $properties = null,
+        ?NameConverterInterface $nameConverter = null,
+    ) {
+        parent::__construct($managerRegistry, $logger, $properties, $nameConverter);
+    }
+
+    protected function filterProperty(string $property, mixed $value, QueryBuilder $queryBuilder, QueryNameGeneratorInterface $queryNameGenerator, string $resourceClass, ?Operation $operation = null, array $context = []): void
+    {
+        if (
+            null === $value
+            || !$this->isPropertyEnabled($property, $resourceClass)
+            || !$this->isPropertyMapped($property, $resourceClass, true)
+        ) {
+            return;
+        }
+
+        $extraProperties = $operation?->getExtraProperties();
+        $resource = $extraProperties['_value'] ?? null;
+        if (!$resource) {
+            $this->logger->warning(\sprintf('No resource found for property "%s".', $property));
+
+            return;
+        }
+
+        $alias = $queryBuilder->getRootAliases()[0];
+        $parameterName = $queryNameGenerator->generateParameterName($property);
+
+        $queryBuilder
+            ->andWhere(\sprintf('%s.%s = :%s', $alias, $property, $parameterName))
+            ->setParameter($parameterName, $resource->getId());
+    }
+
+    /**
+     * {@inheritdoc}
+     */
+    public function getType(string $doctrineType): string
+    {
+        // TODO: remove this test when doctrine/dbal:3 support is removed
+        if (\defined(Types::class.'::ARRAY') && Types::ARRAY === $doctrineType) {
+            return 'array';
+        }
+
+        return match ($doctrineType) {
+            Types::BIGINT, Types::INTEGER, Types::SMALLINT => 'int',
+            Types::BOOLEAN => 'bool',
+            Types::DATE_MUTABLE, Types::TIME_MUTABLE, Types::DATETIME_MUTABLE, Types::DATETIMETZ_MUTABLE, Types::DATE_IMMUTABLE, Types::TIME_IMMUTABLE, Types::DATETIME_IMMUTABLE, Types::DATETIMETZ_IMMUTABLE => \DateTimeInterface::class,
+            Types::FLOAT => 'float',
+            default => 'string',
```

Comment:
An IriFilter would filter on identifiers and relation supported types would be: strings, uuid, integers 

Anyways, IriConverterParameterProvider provides an Object therefore I think that doctrine will make that work without issue, for example:


```
/books?author=/author/1
```

gives 

```
$queryBuilder->andWhere('book.author = :author')->setParameter('author', $value);